src/NServiceBus.Transport.RabbitMQ/Connection/ChannelProvider.cs

```diff
@@ -19,6 +21,35 @@ public ChannelProvider(ConnectionFactory connectionFactory, IRoutingTopology rou
         public void CreateConnection()
         {
             connection = connectionFactory.CreatePublishConnection();
+            connection.ConnectionShutdown += Connection_ConnectionShutdown;
```

Comment:
We probably want to unsubscribe from this event somewhere.

Reply:
Why do you think that is needed?

Reply:
@bording my bad. I've confused the dependency direction between the Connection and the ChannelProvider and thought we might have a memory leak here. All good.